src/bsp/src/stm32/hal/src/uart_print.c

```diff
@@ -53,8 +53,9 @@ void UartPrint_sendTxCallback() {
     ZeroCopyBuff buff = CircularBuff_getZeroCopy(&cbuffUartPrint, UINT16_MAX);
     if (buff.status == CircularBuff_Ret_Ok) {
         HAL_UART_Transmit_IT(HUART_PRINT, (uint8_t*)buff.data, buff.length);
-        lastUartPrintTransferSize = buff.length;
     }
+
+    lastUartPrintTransferSize = buff.length;
```

Comment:
Is it to prevent retries?

Reply:
It's to fix a bug where if there is a problem while getting the buffer, the lastUartPrintSize never returns to 0 and the circular buffer only returns errors from thereon